FactoryWeather/Common/Network/Geo Names/GeoNamesApiManager.swift

```diff
@@ -7,28 +7,27 @@
 //
 
 import Foundation
+import Promises
 
 struct GeoNamesApiManager {
-    static func getLocations(forText text: String, success: @escaping ([Location]) -> Void, failure: @escaping (LocalizedError) -> Void) {
-        guard let url = createUrl(forText: text)
-            else { return DispatchQueue.main.async { failure(NetworkManagerError.urlCreationFailure) } }
-        var request = URLRequest(url: url)
-        request.httpMethod = HTTPRequestMethod.get.rawValue
-        let session = URLSession.shared
-        let task = session.dataTask(with: request) { data, response, error in
-            if let error = error {
-                return DispatchQueue.main.async { failure(NetworkManagerError.generic(error)) }
-            }
-            do {
-                guard let unwrappedData = data
-                    else { return DispatchQueue.main.async { failure(NetworkManagerError.dataUnwrappingFailure) } }
-                let geoName = try JSONDecoder().decode(GeoName.self, from: unwrappedData)
-                return DispatchQueue.main.async { success(geoName.locations) }
-            } catch {
-                return DispatchQueue.main.async { failure(NetworkManagerError.parsingDataFailure) }
-            }
+    static func getLocations(forText text: String) -> Promise<[Location]> {
+        return Promise(on: DispatchQueue.global(qos: .background)) { fulfill, reject in
```

Comment:
It's actually not necessary to publish this `Promise` on a `background` queue, as `URLSession.shared.dataTask` will dispatch on a `non-main` queue by `default`. Regardless, it's a good idea to set it as a precaution.

Reply:
Ok, I used it just to try it out.